pybird/__init__.py

```diff
@@ -693,20 +693,21 @@ def _socket_query(self, query):
         sock.connect(self.socket_file)
         sock.send(query)
 
-        data = b''
+        data = []
 
         while True:
-            if any([data.find(f"\n{code:04}".encode("utf-8")) != -1
+            this_read = sock.recv(1024*1024)
+            data.append(this_read)
+            if len(this_read) > 256:
+                tail = this_read[-256:].decode('utf-8')
+            else:
+                tail = b''.join(data[-2:])[-256:].decode('utf-8')
+            if any([tail.find(f"\n{code:04}") != -1
                 for code in self.error_fields + self.success_fields]):
                 break
-            this_read = sock.recv(1024)
-            if not this_read:
-                self.log.debug(data)
-                raise ValueError("Could not read additional data from BIRD")
-            data += this_read
```

Comment:
this kills the performance and results it a fully utilized CPU core that keeps copying the buffer over and over